ctapipe/core/container.py

```diff
@@ -112,6 +112,11 @@ def validate(self, value):
             # recursively check sub-containers
             value.validate()
 
+        if isinstance(value, Map):
+            for key, map_value in value.items():
```

Comment:
key is not needed right? Or better, catch the validation error and report the key so users can better see what failed?

Reply:
```
try:
   map_value.validate()
except Exception as e:
   raise FieldValidationError(f"Failed to validate map entry {key}: {e}") from e
```

Reply:
Good idea.  Now the resulting error message is for example:
```
FieldValidationError: ArrayEventContainer Field 'dl1': DL1Container Field 'tel': [3]: DL1CameraContainer Field 'parameters': ImageParametersContainer Field 'hillas': HillasParametersContainer Field 'r': the value '10.0 m' (<class 'astropy.units.quantity.Quantity'>) is invalid: : 'm' (length) and 'deg' (angle) are not convertible
```